how-to/workspace-platform-starter/docs/how-to-define-app-launch-preference.md

```diff
@@ -0,0 +1,320 @@
+ */
+export interface HostLaunchOptions {
+ /**
+  * If specified it indicates you do not want to use a browser window for this view but a platform window.
+  */
+ url?: string;
+
+ /**
+  * If specified it indicates a preference to be used by this type of host.
+  */
+ title?: string;
+
+ /** The Icon you would prefer the window shows. */
+ icon?: string;
+
+ /**
+  * Should the header for the content be hidden
+  */
+ hasHeaders?: boolean;
+
+ /**
+  * Should the host support multi layouts (e.g. pages). Assumes the default for each host will be used.
+  */
+ disableMultiplePages?: boolean;
+
+ /**
+  * Should the toolbar options of a window be hidden if they are available?
+  */
+ disableToolbarOptions?: boolean;
+
+ /**
+  * If this host supports multiple layouts what should the layout (e.g page) title be?
+  */
+ pageTitle?: string;
+
+ /**
+  * If this host supports multiple layouts what should the icon be for the layout (e.g. page) be?
+  */
+ pageIcon?: string;
+}
+```
+
+#### View settings
+
+We let you specify the following view specific settings:
+
+- url - Why would you want to override the view url? For views you may wish to provide an updated url that passes a particular query string parameter. You may want to do this from a platform perspective without modifying the view manifest or the inline view (so you can see the difference between the default and the platform override).
+- interop - You may wish to specify that a view is assigned a specific user channel (e.g. green, yellow) either in the app definition (so you don't have to change the inline view setting or the manifest file) or dynamically (you might want to assign a view against a particular user channel given the circumstances at launch).
+- customData - when building a view the view may take advantage of customData to help it support different modes or to prefill certain inputs etc (contextual data sharing or intents are also an option if you want external apps to be able to influence this).
+
+#### Updatable View Options
+
+An app directory is the golden source for the description of an app. A platform can launch application in many ways but it may want to restrict what can be dynamically changed when launching an app and this is where the updatable setting comes in. This is an array of options that contain the name of the setting that can be configured and an optional array of constraints that should be checked before trying to apply the update. We have inlined some of the options below to make it easier to read.
+
+```javascript
+/**
+ * Which Launch Options are updatable and are there any constraints
+ */
+export interface UpdatableViewLaunchPreference {
+ /**
+  * What setting is updatable?
+  */
+ name?: "url" | "custom-data" | "interop" | "bounds" | "centered" | "host-options";
+
+ /**
+  * Is there a constraint that the platform can apply?
+  */
+ constraint?: "url-domain" | "url-page" | "url-any" | "url-none"[];
```

Comment:
Needs some brackets otherwise the type is incorrect
`("url-domain" | "url-page" | "url-any" | "url-none")[]`

Reply:
done in both places.